google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java

```diff
@@ -432,6 +518,13 @@ static LogEntry fromPb(com.google.logging.v2.LogEntry entryPb) {
         builder.setTimestamp(millis);
       }
     }
+    if (entryPb.hasReceiveTimestamp()) {
```

Comment:
Optional. Taking into account that this weird time computation logic is copied in two places now, please consider having a private method for that to avoid duplication.